lib/ansible/module_utils/azure_rm_common.py

```diff
@@ -604,7 +643,7 @@ def network_client(self):
         self.log('Getting network client')
         if not self._network_client:
             self.check_client_version('network', network_client_version, AZURE_EXPECTED_VERSIONS['network_client_version'])
-            self._network_client = NetworkManagementClient(self.azure_credentials, self.subscription_id)
+            self._network_client = NetworkManagementClient(self.azure_credentials, self.subscription_id, base_url=self.base_url)
```

Comment:
@lmazuel thanks for the suggestion! Any thoughts on why I'd be getting `APIVersion 2017-06-01 is not available` with NetworkManagementClient (and possibly the other ones as well)?

`pip freeze | grep network` shows me `azure-mgmt-network==1.0.0`